src/main/typescript/helpers.ts

```diff
@@ -14,3 +18,35 @@ export function toFixed(
 
   return number.toFixed(fractionDigits).replace(/[.,]?0+$/, "");
 }
+
+/** Create the default message data for roll messages. */
+export function createDefaultMessageData(
+  speaker: foundry.data.ChatMessageData["speaker"]["_source"],
+  rollMode: RollMode
+): ChatMessageDataConstructorData {
+  return {
+    speaker,
+    whisper: whisperTargets(rollMode),
+    blind: rollMode === "blindroll"
+  };
+}
+
+export function isRollBlindedForCurrUser(blind: boolean): boolean {
+  if (blind) return !getGame().user?.isGM ?? true;
+  else return false;
+}
+
+export function whisperTargets(
+  rollMode: RollMode
+): StoredDocument<User>[] | null {
+  const self = getGame().user;
+  switch (rollMode) {
+    case "publicroll":
+    case "blindroll":
+      return null;
+    case "gmroll":
+      return ChatMessage.getWhisperRecipients("gm");
+    case "selfroll":
+      return self ? [self] : [];
+  }
```

Comment:
I am not sure if this is correct. This looks like you can't whisper a public roll to a subset of users.

Reply:
Is that a base feature in foundry? Because, as far as I know, a public roll is supposed to be completely public. I've never thought of the possibility of having a roll be private to a certain group of people.

Reply:
Well those are two separate privacy settings. A public message is seen by everyone. A whispered message is only seen by the creator and the message recipients.

A public roll shows the result to everyone that can see the message. A blind roll shows the result only to GMs. So it is a valid combination.